modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -222,6 +194,37 @@ public Page<ObjectDefinition> getObjectDefinitionsPage(
 					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
 	}
 
+	@Override
+	public Page<ObjectDefinition>
+			getObjectFolderByExternalReferenceCodeObjectDefinitionsPage(
+				String externalReferenceCode, String search,
+				Pagination pagination)
+		throws Exception {
+
+		ObjectFolder objectFolder =
+			_objectFolderLocalService.getObjectFolderByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		return SearchUtil.search(
+			_getActions(),
+			booleanQuery -> {
+			},
+			null, com.liferay.object.model.ObjectDefinition.class.getName(),
+			search, pagination,
+			queryConfig -> queryConfig.setSelectedFieldNames(
+				Field.ENTRY_CLASS_PK),
+			searchContext -> {
+				searchContext.setAttribute(Field.NAME, search);
+				searchContext.setAttribute(
+					"objectFolderId", objectFolder.getObjectFolderId());
+				searchContext.setCompanyId(contextCompany.getCompanyId());
+			},
+			null,
+			document -> _toObjectDefinition(
+				_objectDefinitionService.getObjectDefinition(
+					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
```

Comment:
What is your opinion about creating a private method to reuse this code? 
I see that the code is very similar in both usages. Also, we can add conditions in searchContext -> {} to control what we set as attribute (I searched usages of  searchContext -> {} in the portal).